javascript/webdriver/touchsequence.js

```diff
@@ -0,0 +1,280 @@
+ * @return {!webdriver.ActionSequence} A self reference.
+ */
+webdriver.TouchSequence.prototype.release = function(location) {
+  var command = new webdriver.Command(webdriver.CommandName.TOUCH_UP).
+      setParameter('x', location.x).
+      setParameter('y', location.y);
+
+  this.schedule_('touchUp', command);
+  return this;
+};
+
+
+/**
+ * Touch and scroll, moving by xoffset and yoffset.
+ *
+ * @param {xoffset: number, yoffset: number} offset The offset to scroll to.
+ * @return {!webdriver.ActionSequence} A self reference.
+ */
+webdriver.TouchSequence.prototype.scroll = function(offset) {
+  var command = new webdriver.Command(webdriver.CommandName.TOUCH_SCROLL).
+      setParameter('xoffset', offset.xoffset).
+      setParameter('yoffset', offset.yoffset);
+
+  this.schedule_('touchScroll', command);
+  return this;
+};
+
+
+/**
+ * Touch and scroll starting at elem, moving by xoffset and yoffset.
+ *
+ * @param {webdriver.WebElement} elem The element where scroll starts.
+ * @param {xoffset: number, yoffset: number} offset The offset to scroll to.
+ * @return {!webdriver.ActionSequence} A self reference.
+ */
+webdriver.TouchSequence.prototype.scrollFromElement = function(elem, offset) {
+  // The interactions API expect the element ID to be encoded as a simple
+  // string, not the usual JSON object.
+  var id = /** @type {!webdriver.WebElement} */ (elem).getId().
+      then(function(value) {
+        return value['ELEMENT'];
+      });
+
+  var command = new webdriver.Command(webdriver.CommandName.TOUCH_SCROLL).
+      setParameter('element', id).
+      setParameter('xoffset', offset.xoffset).
+      setParameter('yoffset', offset.yoffset);
+
+  this.schedule_('touchScroll', command);
+  return this;
+};
+
+
+/**
+ * Flick, starting anywhere on the screen, at speed xspeed and yspeed.
+ *
+ * @param {xspeed: number, yspeed: number} speed The speed to flick in each
+       direction, in pixels per second.
+ * @return {!webdriver.ActionSequence} A self reference.
+ */
+webdriver.TouchSequence.prototype.flick = function(speed) {
+  var command = new webdriver.Command(webdriver.CommandName.TOUCH_FLICK).
+      setParameter('xspeed', speed.xspeed).
+      setParameter('yspeed', speed.yspeed);
+
+  this.schedule_('touchFlick', command);
+  return this;
+};
+
+
+/**
+ * Flick starting at elem and moving by xoffset and yoffset at specified speed.
+ *
+ * @param {webdriver.WebElement} elem The element where scroll starts.
+ * @param {xoffset: number, yoffset: number} offset The offset to flick to.
+ * @param {number} speed The speed to flick at in pixels per second.
+ * @return {!webdriver.ActionSequence} A self reference.
+ */
+webdriver.TouchSequence.prototype.flickElement = function(elem, offset, speed) {
+  // The interactions API expect the element ID to be encoded as a simple
+  // string, not the usual JSON object.
+  var id = /** @type {!webdriver.WebElement} */ (elem).getId().
```

Comment:
This is repeated a few times and is probably worth adding a dedicated function for (since ActionSequence uses it too).
How about `webdriver.WebElement.prototype.getRawId`?